docs/guides/vqa_guide/introduction.rst

```diff
@@ -0,0 +1,195 @@
+
+.. literalinclude:: ../../examples/guides/vqa_guide.py
+  :language: python
+  :start-after: # VQE initializer
+  :end-before: # --- End
+
+TODO: the example above raises an exception
+
+One can also construct new VQE objects based off the already existing ones. To do so, one can use one of the :meth:`replace_xyz` methods, where :code:`xyz` stands for an attribute you want to replace. The important thing to notice is that all those methods are not mutating an existing object, but creating a new one with one attribute replaced and the others preserved.
+
+We can verify this is indeed the case by inspecting objects returned by :meth:`replace_xyz` methods and comparing them to the original objects. for instance:
+
+.. literalinclude:: ../../examples/guides/vqa_guide.py
+  :language: python
+  :start-after: # replace optimizer
+  :end-before: # --- End
+
+.. code:: text
+
+    Ansatzes equal? True
+    Grouping equal? True
+    Estimation method equal? True
+    Hamiltonians equal? True
+    Optimizers equal? False
+    First optimization method: L-BFGS-B
+    Second optimization method: COBYLA
+
+..  note::
+
+   Arguments to :meth:`replace_xyz` follow the same semantics as the arguments to :code:`orquestra.vqa.algorithms.vqe.VQE.__init__` method. In particular, all preprocessors has to be passed as callables and not strings.
+
+To find all the attributes of :class:`VQE <orquestra.vqa.algorithms.vqe.VQE>` that can be replaced in this way, refer to the class documentation.
+
+TODO: Add some good initial references?
+
+Implementations of QAOA and QCBM algorithms follow the same design principles as VQE. In particular:
+
+* Their instances can be constructed using either simplified :meth:`default` method, or with more control using their initializer
+* They store optimizer as one of the attributes. The optimization is done by calling
+  :meth:`find_optimal_params`
+* The instances are supposed to be treated as immutable, but new instances can be constructed by replacing respective attributes using :meth:`replace_xyz` methods.
+
+Knowing those similarities, in the rest of the introduction we will only highlight differences between Orquestra's implementation of VQE and QAOA/QCBM.
+
+QAOA
+----
+
+The Quantum Approximate Optimization Algorithm is implemented by the :class:`QAOA <orquestra.vqa.algorithms.QAOA>` class. The  :meth:`default <orquestra.vqa.algorithms.QAOA.default>` method constructs :class:`QAOA` object using Fahri ansatz with specified number of layers. Using class' initializer, one can further customize what ansatz is used.
```

Comment:
I think it requires more details – e.g. QAOA doesn't have grouping and VQE does.
At least describe what are the inputs for default are and why some are missing from QAOA.